### DISCUSSION TITLE
[FEATURE/SUGGESTION] HIGH-RESOLUTION 3D RENDERING

### DISCUSSION THREAD

**Discussion by eudavibrito** [rafaelvcaetano/melonDS-android#1248] [Ideas]
Hello! Firstly, I want to express my gratitude to the creator and supporters of this project for the INCREDIBLE work you're doing. It's truly comforting to see individuals actively dedicated to developing a Nintendo DS emulator for Android. Your efforts mean a lot to the community.

I'm reaching out to discuss and potentially suggest that the developer consider prioritizing the addition of support for high-resolution (HQ/3D) game processing, if feasible. While this functionality exists in NDS emulators on PC, it hasn't been explored in this project yet. For me, it holds great significance as the visual and graphic aspects of gameplay are integral to the overall experience. Currently, I believe this feature is the only missing element for the emulator to be truly perfect.

I'd love to engage in a discussion with the developer to understand if there are any intentions to incorporate this support into the project in the near future.

**Comment by rafaelvcaetano** [comment]
You mean like this? It's coming in the next update.  
![image](https://github.com/rafaelvcaetano/melonDS-android/assets/9086379/d1c3fcd7-b28d-4333-afc6-b166ebedeb91)

**Comment by eudavibrito** [comment]
Exactly!!! The regular resolution, which I believe is the same as the original NDS, has that jagged/pixelated effect on the lines that even reminds me a bit of Game Boy Advance graphics. In 3D games, it causes considerable discomfort, as in the game you used as an example.

Thank you so much for taking care of this matter. I'm truly very happy and excited to play with a more beautiful and clean image resolution in the next update, even with the addition of the SMAA filter, which I believe that may come with the new screen filters in the future.
